e2e_samples/temperature_events/Infra/terraform/modules/functions/main.tf

```diff
@@ -0,0 +1,31 @@
+data "azurerm_resource_group" "rg" {
+  name = var.resource_group_name
+}
+
+resource "azurerm_storage_account" "storage" {
+  name                     = replace("st-${var.function_name}-${data.azurerm_resource_group.rg.tags.resource_name}", "-", "")
+  location                 = data.azurerm_resource_group.rg.location
+  resource_group_name      = data.azurerm_resource_group.rg.name
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+  min_tls_version          = "TLS1_2"
+}
+
+resource "azurerm_function_app" "function_app" {
+  name                       = "func-${var.function_name}-${var.resource_name}"
+  location                   = data.azurerm_resource_group.rg.location
+  resource_group_name        = data.azurerm_resource_group.rg.name
+  app_service_plan_id        = var.appservice_plan
+  storage_account_name       = azurerm_storage_account.storage.name
+  storage_account_access_key = azurerm_storage_account.storage.primary_access_key
+  os_type                    = "linux"
+
+  version    = "~3"
+  https_only = true
+
+  identity {
+    type = "SystemAssigned"
+  }
+
+  app_settings = var.appsettings
```

Comment:
I'd like to propose another approach here. 

Since you've already setup a `Managed Identity`, how about constructing the `app_settings` string here as we've done previously? 

Due to the `Managed Identity` It shouldn't be necessary to include the secret in the Key Vault URI. Just create a new `azurerm_key_vault_access_policy` for the `function_app`'s Identity.

Reply:
Constructing app settings here is a bit more difficult because we need to create eventhubs before this but azure functions do not have access to eventhub connections yet.